src/control/client/config.go

```diff
@@ -0,0 +1,82 @@
+//
+// (C) Copyright 2018-2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package client
+
+import (
+	"io/ioutil"
+	"strconv"
+	"github.com/daos-stack/daos/src/control/log"
+)
+
+const (
+	providerEnvKey = "CRT_PHY_ADDR_STR"
+	ofi_interface_env = "OFI_INTERFACE"
+	ofi_port_env = "OFI_PORT"
+	daos_agent_drpc_sock_env = "DAOS_AGENT_DRPC_SOCK"
+)
+
+func (c *ClientConfiguration) LoadConfig() error {
+	bytes, err := ioutil.ReadFile(c.Path)
+	if err != nil {
+		return err
+	}
+	if err = c.parse(bytes); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func ProcessEnvOverrides(c *ClientConfiguration) int {
```

Comment:
why not make this a method on the ClientConfiguration type to be consistent. this processing of Env overrides probably makes sense for the client but we need to be aware that it's the opposite of what we are doing in the server

Reply:
Good idea.